feat(mkdocs): add automatic documentation using mkdocs

SauravMaheshkar opened the pull request:
Fixes #16.

Ideally this should be worked on after #13 is merged. Following that let's rebase and then proceed with adding docstrings & typehints to `restorers/`.

SauravMaheshkar committed:
feat(mkdocs): add initial template for mkdocs

SauravMaheshkar committed:
Merge branch 'main' into saurav/add-auto-docs

SauravMaheshkar committed:
docs: add initial docs template

SauravMaheshkar commented:
@soumik12345 If you could checkout to this branch and run

```lang-bash
mkdocs serve
```

soumik12345 commented:
> @soumik12345 If you could checkout to this branch and run
> 
> ```
> mkdocs serve
> ```

Hi @SauravMaheshkar I am getting the following error:

```
Error: Config file 'mkdocs.yml' does not exist.
```

ariG23498 commented:
Wouldn't it be better if we could review the documentation on the website (restorer docs) rather than here?

soumik12345 commented:
> Wouldn't it be better if we could review the documentation on the website (restorer docs) rather than here?

soumik12345 commented:
> Wouldn't it be better if we could review the documentation on the website (restorer docs) rather than here?

@ariG23498 you can, just hit

```bash
pip install -q mkdocs mkdocstrings mkdocstrings-python
mkdocs serve
```

soumik12345 committed:
Merge branch 'main' into saurav/add-auto-docs

Samarendra109 committed:
Documentation: Added documentation for blocks defined for NAFNet implementation

Samarendra109 committed:
Apply suggestions from code review

Dos string change suggestions from @soumik12345 was accepted.

Co-authored-by: Soumik Rakshit <19soumik.rakshit96@gmail.com>

Samarendra109 committed:
Updated type hints, added type hints to doc string and removed unnecessary lines from docstrings

Samarendra109 committed:
Updated type hints, added type hints to doc string and removed unnecessary lines from docstrings

Samarendra109 committed:
Merged changes

Samarendra109 committed:
Fixed Linting

Samarendra109 committed:
Fixed Import

soumik12345 committed:
Documentation: Added documentation for blocks defined for NAFNet implementation (#66)

Fixes #61 

Changes proposed by this PR :-

Added references to the doctrings for
[SimpleGate](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafblock.py#L11).
Elaborated docstrings for
[ChannelAttention](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafblock.py#L37)
with references and more details.
Elaborated docstrings for
[SimplifiedChannelAttention](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafblock.py#L71)
with references and more details.
Elaborated docstrings for
[NAFBlock](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafblock.py#L100)
with references and more details.
Added references and parameters to the docstrings for
[PixelShuffle](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafnet.py#L10).
Added parameters to the docstrings for
[UpScale](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafnet.py#L39).
Added references to the docstrings for
[NAFNet](https://github.com/TrystAI/restorers/blob/main/restorers/model/nafnet/nafnet.py#L85).

Request for Review: @soumik12345 @SauravMaheshkar

soumik12345 committed:
add: doctrings for BaseEvaluator

soumik12345 committed:
add: documentation for BaseEvaluator

soumik12345 committed:
fix: linting

soumik12345 committed:
add: documentation for LoLEvaluator

soumik12345 committed:
update: remove requirements.txt + add optional dependencies for dev, test and docs

soumik12345 committed:
update: workflow

soumik12345 committed:
fix: linting + pyproject.toml

soumik12345 committed:
update: pyproject.toml

soumik12345 committed:
docs: Add docstrings for restorers.evaluation module (#67)

Fixes https://github.com/TrystAI/restorers/issues/63

Changes proposed by this PR:

- [x] Add docstrings for `BaseEvaluator`
- [x] Add docstrings for `LoLEvaluator`
- [x] Removed `requirements.txt`
- [x] Added optional dependencies for `dev`, `test`, and `docs`

soumik12345 committed:
update: docstrings for losses + update zero-dce arxiv link

soumik12345 committed:
update: added return sections to docstrings for zero-reference losses

soumik12345 committed:
update: PR template

soumik12345 committed:
fix: linting

soumik12345 committed:
fix: linting

soumik12345 committed:
fix: latex in docstring

soumik12345 committed:
update: NAFBlock docstring

soumik12345 committed:
fix: latex in docstrings

soumik12345 committed:
add: documentation for latex

soumik12345 committed:
update: fix escape character issues in docstrings

soumik12345 committed:
add: mkdocs plugins for search and minify

soumik12345 committed:
add: docstrings for inference module

soumik12345 committed:
add: documentation for inference module

soumik12345 committed:
update: mkdocs.yml

soumik12345 committed:
fix: linting

soumik12345 committed:
add: docstrings and documentations for dataloader module

soumik12345 committed:
fix: linting

soumik12345 committed:
update: refactored recursive residual groups + corresponding unit tests

soumik12345 committed:
add: MirNetV2 documentation

soumik12345 committed:
update: add diagrams for MrNetv2

soumik12345 committed:
update: refactor FastZeroDCE implementation

soumik12345 committed:
fix: linting

soumik12345 committed:
fix: zero-dce unit tests

soumik12345 committed:
fix: fast_zero_dce import

soumik12345 committed:
update: refactored PixelShuffle and UpScale

soumik12345 committed:
fix: minor bug at UpScale

soumik12345 committed:
fix: reference link in docstring for FastZeroDce

soumik12345 committed:
update: docstrings + documentation for nafnet

soumik12345 committed:
update: added reference for nafnet pytorch implementation

soumik12345 committed:
update: index.md

Samarendra109 committed:
Fixed documentation for NAFNet

Samarendra109 committed:
Fixed more issues in documentation for NAFNet

Samarendra109 committed:
Fixed linting

soumik12345 committed:
docs: Add documentation for the dataloader, model, losses and inference module (#68)

Fixes https://github.com/TrystAI/restorers/issues/62
Fixes https://github.com/TrystAI/restorers/issues/64
Fixes https://github.com/TrystAI/restorers/issues/60

Changes proposed by this PR:

- [x] Fix docstrings for the `losses` module as per
https://github.com/TrystAI/restorers/issues/62
- [x] Add documentation for the `losses` module
- [x] Add documentation for the `inference` module as per
https://github.com/TrystAI/restorers/issues/64
- [x] Add documentation for the `dataloader` module as per
https://github.com/TrystAI/restorers/issues/60
- [x] Add documentation for the `model.mirnetv2` module
- [x] Add documentation for the `model.nafnet` module
- [x] Add documentation for the `model.zero_dce` module
- [x] Update PR template

ariG23498 commented on restorers/model/nafnet/nafblock.py:
```diff
@@ -99,32 +129,59 @@ def get_config(self) -> dict:
+    <<<<<<< HEAD
+    =======
+            input_channels (Optional[int]): number of channels in the input (as NAFBlock retains the input size in the output)
+    >>>>>>> ea3089f6eb211c37a2a28eb183312fd8a0c1d106
```
I think this is a merge conflict @soumik12345

soumik12345 replied:
cc: @Samarendra109

soumik12345 committed:
add: workflow for deploying documentation

soumik12345 committed:
update: workflows

soumik12345 committed:
update: docstrings

soumik12345 committed:
update: added notebook examples for training models

soumik12345 committed:
update: docstrings

soumik12345 committed:
update: documentation + worflows

soumik12345 committed:
refactor: experiments

soumik12345 committed:
update: documentation + docstrings

soumik12345 committed:
update: documentation

soumik12345 committed:
update: packaging

soumik12345 committed:
update: packaging

soumik12345 committed:
docs: Deploy (#69)

Changes proposed by this PR:

- [x] Add workflow for deploying documentation upon push into `main` or
`saurav/add-auto-docs`.
- [x] Parallelize unit test workflow
- [x] Add example notebooks to documentation.

soumik12345 committed:
update: docs

soumik12345 committed:
refactor: UnsupervisedLOLDataLoader

soumik12345 committed:
fix: minor bug in LOLDataLoader + UnsupervisedLOLDataLoader

soumik12345 committed:
fix: minor bug in LOLDataLoader + UnsupervisedLOLDataLoader

soumik12345 committed:
update: optimize imports

soumik12345 committed:
add: github action to automatically publish to testpypi

soumik12345 committed:
add: pre-commit

soumik12345 committed:
update: embedded wandb report in docs

soumik12345 committed:
fix: dataset size in UnsupervisedLOLDataLoader

soumik12345 committed:
add: LargeLowLightDataset

soumik12345 committed:
update: LargeLowLightDataset with max_images

soumik12345 committed:
Update pyproject.toml

Co-authored-by: Aritra Roy Gosthipaty <aritra.born2fly@gmail.com>

soumik12345 committed:
add: authors name and email to pyproject.toml

soumik12345 committed:
update: target branches for pypi-publish and docs + changed testpypi to pypi

soumik12345 committed:
refactor: Inheritance Structure of UnsupervisedLOLDataLoader (#70)

Changes proposed by this PR:

- [x] Add `UnsupervisedLowLightDatasetFactory` subclassed from
`LowLightDatasetFactory`.
- [x] Make `UnsupervisedLOLDataLoader` inherit from
`UnsupervisedLowLightDatasetFactory`.
- [x] Update documentation accordingly.
- [x] Add a workflow to automatically publish to testpypi. Will be
changed to Pypi and the target branch to `main` once PR is reviewed and
approved.
- [x] Added pre-commit hooks to automatically run specific checks on
commit.
- [x] Added implementation of `LargeLowLightDataset`.
